Week 1. Chat-api

anton0701 opened the pull request:
Chat-api. grpc, домашка первой недели

anton0701 committed:
chat proto implemented

anton0701 committed:
log + using getters instead of fields

Shemistan commented on go.mod:
```diff
@@ -3,8 +3,16 @@ module github.com/anton0701/chat-server
 go 1.20
```
почему без билдов? На сколько я помню еще github actions было на первой неделе

anton0701 replied:
вроде в первой неделе нет github actions

Shemistan commented on grpc/api/chat_v1/chat.proto:
```diff
@@ -0,0 +1,33 @@
+  rpc SendMessage(SendMessageRequest) returns (google.protobuf.Empty);
+}
+
+message CreateRequest {
```
usernames - это уникально поле? Может тут должен быть email или id
и преполагаешь ли ты, что у чата будет какое-то название?

anton0701 replied:
> преполагаешь ли ты, что у чата будет какое-то название?

думаю, что у чата должно быть название, но этого поля нет в описании домашки

> usernames - это уникально поле? Может тут должен быть email или id

согласен, сюда бы лучше айдишники пользователей передавать, но я делал по ТЗ из домашки

в итоге, как нужно сделать? как мне кажется более правильным или по ТЗ?

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+
+import (
+	"context"
+	"fmt"
```
Сортировка - написал в другом pr

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+	"strings"
+)
+
+const grpcPort = 50052
```
объедени

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+	desc.UnimplementedChatV1Server
+}
+
+func (s *server) Create(ctx context.Context, req *desc.CreateRequest) (*desc.CreateResponse, error) {
```
пока не используешь context - напиши  "_  context.Context"

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+
+func (s *server) Create(ctx context.Context, req *desc.CreateRequest) (*desc.CreateResponse, error) {
+	log.Println(grpcChatApiDesc)
+	log.Printf("Method Create. Input params:\nUsernames: \n%s\n************\n\n", strings.Join(req.GetUsernames(), "\n"))
```
объедени

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+	log.Println(grpcChatApiDesc)
+	log.Printf("Method Create. Input params:\nUsernames: \n%s\n************\n\n", strings.Join(req.GetUsernames(), "\n"))
+
+	resp := &desc.CreateResponse{
```
лишнее

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+	return resp, nil
+}
+
+func (s *server) Delete(ctx context.Context, req *desc.DeleteRequest) (*emptypb.Empty, error) {
```
ctx

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+
+func (s *server) Delete(ctx context.Context, req *desc.DeleteRequest) (*emptypb.Empty, error) {
+	log.Println(grpcChatApiDesc)
+	log.Printf("Method Delete. Input params:\nId: %d\n************\n\n", req.GetId())
```
объедени

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+	log.Println(grpcChatApiDesc)
+	log.Printf("Method Delete. Input params:\nId: %d\n************\n\n", req.GetId())
+
+	return &emptypb.Empty{}, nil
```
тут уже ок

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+	return &emptypb.Empty{}, nil
+}
+
+func (s *server) SendMessage(ctx context.Context, req *desc.SendMessageRequest) (*emptypb.Empty, error) {
```
ctx

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+
+func (s *server) SendMessage(ctx context.Context, req *desc.SendMessageRequest) (*emptypb.Empty, error) {
+	log.Println(grpcChatApiDesc)
+	log.Printf("Method SendMessage. Input params:\nFrom: %s\nText: %s\nTimestamp: %s************\n\n", req.GetFrom(), req.GetText(), req.GetTimestamp())
```
объедени

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -0,0 +1,62 @@
+	return &emptypb.Empty{}, nil
+}
+
+func main() {
```
main вынести на верх, это точка входа и искать его по файлу - не лучшее занятие

anton0701 replied:
++

anton0701 committed:
fix after PR review

anton0701 committed:
api-methods, requests, responses renamed: "Chat" added
CreateChatRequest: fields "UserIds", "ChatName", "ChatDescription" added; field "usernames" removed

anton0701 committed:
аббревиатуры в названии переменных написал заглавными буквами
